packages/vsx-registry/src/common/vsx-version.ts

```diff
@@ -0,0 +1,41 @@
+/********************************************************************************
+ * Copyright (C) 2022 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+export type VSXExtensionVersion = number[];
+
+export function parseVersion(versionString: string): VSXExtensionVersion | undefined {
+    const array = versionString.split('.').map(e => Number(e));
+    return array.some(isNaN) ? undefined : array;
+}
+
+export function compareVersion(a: VSXExtensionVersion, b: VSXExtensionVersion): (-1 | 0 | 1) {
```

Comment:
@vince-fugnitto, this looks like it's recreating some of the work you recently wrote tests for - is there already a way to do this?

Reply:
I use `semver` which should handle most of the use-cases we have included in the changes I believe.

Ex:
https://github.com/eclipse-theia/theia/blob/dd900be938f6981d2a08edb60e42151d96804ec0/dev-packages/ovsx-client/src/ovsx-client.ts#L194-L209

Reply:
@vince-fugnitto Thanks for the pointer, I implemented the comparison using `semver` and removed the custom implementation 👍